src/App.tsx

```diff
@@ -15,36 +17,100 @@
   'Garlic',
 ];
 
+let goods = [...goodsFromServer];
+let isReverseCLicked = false;
+
 export const App: React.FC = () => {
+  const ALPHABETICALLY: string = 'alphabetically';
+  const BY_LENGTH: string = 'by length';
+  const RESET: string = 'reset';
+  const [sortGoods, setSortGoods] = useState('');
+  const [reverse, setReversed] = useState(false);
+
+  function setSort() {
+    if (sortGoods === ALPHABETICALLY) {
+      goods = [...goodsFromServer].sort((goodA, goodB) =>
+        goodA.localeCompare(goodB),
+      );
+    }
+
+    if (sortGoods === BY_LENGTH) {
+      goods = [...goodsFromServer].sort(
+        (goodA, goodB) => goodA.length - goodB.length,
+      );
+    }
+
+    if (reverse) {
+      goods.reverse();
+    }
+  }
+
+  if (sortGoods === RESET) {
+    goods = [...goodsFromServer];
+  }
+
+  if (isReverseCLicked) {
+    goods.reverse();
+    isReverseCLicked = false;
+  } else {
+    setSort();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortGoods !== ALPHABETICALLY,
+          })}
+          onClick={() => setSortGoods(ALPHABETICALLY)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortGoods !== BY_LENGTH,
+          })}
+          onClick={() => setSortGoods(BY_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': reverse === false,
+          })}
+          onClick={() => {
+            setReversed(!reverse);
+            isReverseCLicked = true;
+          }}
+        >
```

Comment:
[UX] Consider adding a `disabled` attribute to the button when `reverse` is `true`.